### ISSUE TITLE
Integrated output display in notebooks

### ISSUE THREAD

**Issue by amrita112** [napari/napari#990]
## Question

In windows, if a keybinding function in a jupyter notebook prints output, it doesn't show up in the notebook. 

## To Reproduce

1. In windows, open a jupyter notebook and open a napari console
2. Write a keybinding function to display some output using print()
3. Keypress doesn't give output in the notebook

## Expected behavior

Output in the cell in which the keybinding function is defined

## Environment

napari: 0.2.11
Platform: Windows-10-10.0.17763-SP0
Python: 3.7.4 (default, Aug 9 2019, 18:34:13) [MSC v.1915 64 bit (AMD64)]
Qt: 5.9.6
PyQt5: 5.9.2
VisPy: 0.6.4
NumPy: 1.18.1
SciPy: 1.3.1
scikit-image: 0.15.0
Dask: 2.5.2

GL version: 4.6.0 NVIDIA 436.48
MAX_TEXTURE_SIZE: 32768

**Comment by sofroniewn** [comment]
Thanks for reporting this @amrita112, maybe one of the @napari/core-devs that has access to a windows machine can look into replicating this

**Comment by jni** [comment]
@sofroniewn I don't think this is windows-specific, but rather a limitation of notebooks. In a notebook, once a cell is done executing, you can't modify its output[^1]. So as far as know there is no way for us to resolve this appropriately. Packages like Bokeh and streamz do crazy/fun things like display an html/js widget that is synced with a value in Python that can be modified in an independent thread. But imho that is beyond napari's scope, at least for now.

The solution to this short term would be to:
- add a "console" widget that displays stdout and stderr (this is different from the interactive QtConsole)
- improve our documentation to explain where print output goes.

[^1]: At least, not without some significant hacking of the notebook machinery, I think.
